src/Query/Builder.php

```diff
@@ -703,7 +703,14 @@ public function delete($id = null)
         $wheres = $this->compileWheres();
         $options = $this->inheritConnectionOptions();
 
-        $result = $this->collection->deleteMany($wheres, $options);
+        if ($this->limit) {
+            if ($this->limit !== 1) {
```

Comment:
Should `limit === 0` be supported? Or would Eloquent expect that to be a NOP?

I suppose the notion of `limit: 0` meaning "delete all" is an implementation detail of MongoDB's [`delete` command](https://www.mongodb.com/docs/manual/reference/command/delete/#std-label-deletes-array-limit) and not a general concept, so I'm fine with not supporting that.

I do wonder if we should have a separate exception message if `0` is specified that advises users to specify `null` instead (or not call `limit()` at all), as this message might confuse a user with more experience with MongoDB than Eloquent.

Reply:
You're absolutely right. I've also banned 0.